cmd/goal/application.go

```diff
@@ -266,6 +294,22 @@ func processAppInputFile() (args [][]byte, accounts []string, foreignApps []uint
 	return parseAppInputs(inputs)
 }
 
+func splitAppArgsByComma() {
+	var newAppArgs []string
+
+	for i := 0; i < len(appArgs); i++ {
+		if !strings.Contains(appArgs[i], "abi:") && len(appArgs[i]) > 0 {
+			splitByComma := strings.Split(appArgs[i], ",")
+			newAppArgs = append(newAppArgs, splitByComma...)
+		} else if strings.Contains(appArgs[i], "abi:") {
+			// TODO consider case when abi packed value comma with other packed value
+			newAppArgs = append(newAppArgs, appArgs[i])
+		}
+	}
+
```

Comment:
Since I changed the `StringSliceVar` to`StringArrayVar`, we need to process the `appArgs` beforehand by:
- Filter out empty strings.
- Split up concatenated values of form like `str:string1,integer:1337`.
- Allowing abi values to be added to `appArgs`.

One thing I am not sure is that, do we want to process strings containing `abi:` (e.g., split up the string into slices)? Since the parsing will be quite complex by my estimation.